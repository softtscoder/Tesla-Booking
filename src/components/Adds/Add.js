import PropTypes from 'prop-types';
import './Add.css';

function Add(props) {
  const {
    id,
    model,
    color,
    img,
    range,
    motor,
    acceleration,
    price,
    duration,
  } = props;

  return (
    <div className="add" id={id}>
      <img src={img} alt={model} />
      <h2>{model}</h2>
      <p>{color}</p>
      <p>{range}</p>
      <p>{motor}</p>
      <p>{acceleration}</p>
      <p>{price}</p>
      <p>{duration}</p>
      <button type="button">Add to cart</button>
    </div>
  );
}

Add.propTypes = {
  id: PropTypes.number.isRequired,
  model: PropTypes.string.isRequired,
  color: PropTypes.string.isRequired,
  img: PropTypes.string.isRequired,
  range: PropTypes.number.isRequired,
  motor: PropTypes.string.isRequired,
  acceleration: PropTypes.number.isRequired,
  price: PropTypes.number.isRequired,
  duration: PropTypes.number.isRequired,
};

export default Add;