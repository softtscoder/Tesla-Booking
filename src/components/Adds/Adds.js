import TextField from '@mui/material/TextField';
import Button from '@mui/material/Button';

export default function Adds() {
  const handleSubmit = async (e) => {
    console.log("I'm here");
    e.preventDefault();
    const response = await fetch('http://127.0.0.1:3000/cars/create', {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify({
        model: e.target.model.value,
        color: e.target.color.value,
        image: e.target.image.value,
        range: e.target.range.value,
        motor: e.target.motor_type.value,
        acceleration: e.target.acceleration_time.value,
        price: e.target.booking_price.value,
        duration: e.target.booking_duration.value,
      }),
    });
    console.log('response', response);
  };

  return (
    <div>
      <form
        id="form"
        autoComplete="off"
        onSubmit={handleSubmit}
      >
        <TextField
          id="standard-basic"
          label="Model"
          variant="outlined"
          type="text"
          name="model"
          placeholder="Model"
        />
        <TextField
          id="standard-basic"
          label="Color"
          variant="outlined"
          type="text"
          name="color"
          placeholder="Color"
        />
        <TextField
          id="standard-basic"
          label="Image"
          variant="outlined"
          type="text"
          name="image"
          placeholder="https://"
        />
        <TextField
          id="standard-basic"
          label="Range"
          variant="outlined"
          type="text"
          name="range"
          placeholder="Range"
        />
        <TextField
          id="standard-basic"
          label="Motor"
          variant="outlined"
          type="text"
          name="motor_type"
          placeholder="Motor"
        />
        <TextField
          id="standard-basic"
          label="Acceleration"
          variant="outlined"
          type="text"
          name="acceleration_time"
          placeholder="Acceleration"
        />
        <TextField
          id="standard-basic"
          label="Price"
          variant="outlined"
          type="text"
          name="booking_price"
          placeholder="Price"
        />
        <TextField
          id="standard-basic"
          label="Duration"
          variant="outlined"
          type="text"
          name="booking_duration"
          placeholder="Duration"
        />
        <TextField
          id="outlined-multiline-static"
          label="Description"
          multiline
          rows={4}
        />
        <Button type="submit" variant="contained" color="primary">Submit</Button>
      </form>
    </div>
  );
}
