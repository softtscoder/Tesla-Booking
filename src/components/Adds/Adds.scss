.container,
#form {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  width: 100%;
  height: 100%;
  background-color: #f5f5f5;
  padding: 0 20px;
  box-sizing: border-box;
  gap: 10px;
}

.input {
  display: grid;
  grid-template-columns: 1fr 1fr;
  gap: 10px;
}
