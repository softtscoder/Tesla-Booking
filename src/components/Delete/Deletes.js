import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { getCars } from '../../redux/cars/cars';

const Reserve = () => {
  const dispatch = useDispatch();
  const carsData = useSelector((state) => state.cars);

  useEffect(() => {
    dispatch(getCars());
  }, [dispatch]);

  const deleteCar = async (e) => {
    e.preventDefault();
    const response = await fetch(`'http://127.0.0.1:3000/cars/${e.target.id}`, {
      method: 'DELETE',
      headers: { 'Content-Type': 'application/json' },
    });
    response.json();
    if (response.status === 200) {
      window.location.reload();
    }
    return response;
  };

  const formOptions = () => {
    const options = [];
    carsData.forEach((car) => {
      options.push(<option key={car.id} value={car.id}>{car.model}</option>);
    });
    return options;
  };

  return (
    <section id="reservation-form">
      <h2> RESERVATION FORM </h2>
      <form onSubmit={(e) => { deleteCar(e); }}>
        <select name="model" id="reservation-form__model" form="reserve-car">
          {formOptions()}
        </select>
        <button type="submit">DELETE</button>
      </form>
    </section>
  );
};

export default Reserve;
