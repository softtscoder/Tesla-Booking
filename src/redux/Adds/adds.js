const url = 'http://localhost:3000/users';
const GETCARS = 'GETCARS';
const ADDCAR = 'ADDCAR';

function carReducer(state = [], action) {
  switch (action.type) {
    case GETCARS:
      return [...action.cars];
    case ADDCAR:
      return state.map((item) => (
        item.id === action.payload ? { ...item, isAdded: true } : item
      ));
    default:
      return state;
  }
}

const GetCar = (cars) => ({
  type: ADDCAR,
  cars,
});

export const AddCar = (id) => ({
  type: ADDCAR,
  payload: id,
});

export const FetchCars = () => (dispatch) => {
  fetch(url)
    .then((res) => res.json())
    .then((result) => {
      const Data = Object.entries(result);
      Data.forEach((element) => {
        List.push({
          id: element[0].id,
          model: element[0].model,
          color: element[0].color,
          img: element[0].img,
          range: element[0].range,
          motor: element[0].motor,
          acceleration: element[0].acceleration,
          price: element[0].price,
          duration: element[0].duration,
        });
      });
      dispatch(GetCar(List));
    });
};

export default carReducer;
