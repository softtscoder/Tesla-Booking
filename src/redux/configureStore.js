import { configureStore } from '@reduxjs/toolkit';
import detailsReducer from './cars/car-details';
import carsReducer from './cars/cars';
import reservationsReducer from './reservations/reservations';

const store = configureStore({
  reducer: {
    cars: carsReducer,
    details: detailsReducer,
    reservations: reservationsReducer,
  },
});

export default store;
