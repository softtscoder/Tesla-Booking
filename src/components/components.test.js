import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import configureStore from '../redux/configureStore';
import Adds from '../components/Adds/Adds';
import Reserve from '../components/Reserve Form/ReserveForm';
import Deletes from '../components/Delete/Deletes';

describe('Test components', () => {

  test('Reserve component', () => {
    const store = configureStore;
    const component = renderer.create(
      <Provider store={store}>
        <MemoryRouter>
          <Reserve />
        </MemoryRouter>
      </Provider>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Deletes component', () => {
    const store = configureStore;
    const component = renderer.create(
      <Provider store={store}>
        <MemoryRouter>
          <Deletes />
        </MemoryRouter>
      </Provider>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
