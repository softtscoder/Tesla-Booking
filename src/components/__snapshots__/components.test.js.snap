// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test components Deletes component 1`] = `
<section
  id="reservation-form"
>
  <h2>
     DELETE FORM 
  </h2>
  <form
    onSubmit={[Function]}
  >
    <select
      form="reserve-car"
      id="reservation-form__model"
      name="model"
    />
    <button
      type="submit"
    >
      DELETE
    </button>
  </form>
</section>
`;

exports[`Test components Reserve component 1`] = `
<section
  id="reservation-form"
>
  <h2>
     RESERVATION FORM 
  </h2>
  <form
    onSubmit={[Function]}
  >
    <select
      form="reserve-car"
      id="reservation-form__model"
      name="model"
    />
    <input
      id="reservation-form__booking-date"
      max="2030-12-31"
      min="2022-11-1"
      name="reservationDate"
      required={true}
      type="date"
    />
    <button
      type="submit"
    >
      RESERVE
    </button>
  </form>
</section>
`;
